api/src/services/connected-user-log.js

```diff
@@ -0,0 +1,47 @@
+const moment = require('moment');
+const db = require('../db');
+
+const LOG_TYPE = 'connected-user-';
+const TIME_INTERVAL = 30 * 60 * 1000; // 30 minutes
+
+const saveLog = (log) => {
+  if (!log || !log.user) {
+    const error = new Error('Error when saving log: Log Information missing');
+    return Promise.reject(error);
+  }
+  const id = LOG_TYPE + log.user;
+  return db.medicLogs
+    .get(id)
+    .catch(error => {
+      if (error.status === 404) {
+        return { _id: id };
+      }
+      throw error;
+    })
+    .then(doc => {
+      if (log.timestamp - doc.timestamp < TIME_INTERVAL) {
+        return;
+      }
+      const logDoc = Object.assign(doc, log);
+      return db.medicLogs.put(logDoc);
+    });
+};
+
+const getLogs = (interval) => {
+  const options = {
+    startkey: LOG_TYPE,
+    endkey: LOG_TYPE + '\ufff0',
+    include_docs: true
+  };
+  const earliestTimestamp = moment().subtract(interval, 'days').valueOf();
+  return db.medicLogs
+    .allDocs(options)
+    .then((result) => result.rows
+      .filter(row => row.doc.timestamp > earliestTimestamp)
+      .map(row => row.doc));
```

Comment:
I'm concerned about performance here, particularly as we think about scaling up to national deployments with 10s of thousands of users. Specifically it's the `include_docs: true` bit which is the killer here - if you can capture all the information you need in the ID then you never need to actually get the docs which will be a huge win.

I can't think of a way to include the date in the ID, but perhaps you could include a new CouchDB view which means you can search by date, group by user, and use a reduce to return the count. That way all the work is being done on CouchDB which caches the view results so if it's called repeatedly by a monitoring system it won't overload API. This view is using a few advanced features of CouchDB to feel free to reach out if you need more specific tips!